python_modules/dagster/dagster/_core/storage/schedules/sql_schedule_storage.py

```diff
@@ -509,6 +509,22 @@ def get_auto_materialize_asset_evaluations(
             rows = db_fetch_mappings(conn, query)
             return [AutoMaterializeAssetEvaluationRecord.from_db_row(row) for row in rows]
 
+    def get_auto_materialize_evaluations_by_evaluation_id(
+        self, evaluation_id: int
+    ) -> Sequence[AutoMaterializeAssetEvaluationRecord]:
+        with self.connect() as conn:
+            query = db_select(
+                [
+                    AssetDaemonAssetEvaluationsTable.c.id,
+                    AssetDaemonAssetEvaluationsTable.c.asset_evaluation_body,
+                    AssetDaemonAssetEvaluationsTable.c.evaluation_id,
+                    AssetDaemonAssetEvaluationsTable.c.create_timestamp,
+                ]
+            ).where(AssetDaemonAssetEvaluationsTable.c.evaluation_id == evaluation_id)
+
+            rows = db_fetch_mappings(conn, query)
+            return [AutoMaterializeAssetEvaluationRecord.from_db_row(row) for row in rows]
```

Comment:
If it becomes too expensive to deserialize all the records at scale, the og plan was to use something like `select sum(num_requested), sum(num_skipped), sum(num_discarded) where evaluation_id=...` to give summary info